app/src/ui/publish-repository/publish.tsx

```diff
@@ -63,7 +64,7 @@ export class Publish extends React.Component<IPublishProps, IPublishState> {
 
     const publishSettings = {
       name: props.repository.name,
-      description: '',
+      description: getGitDescription(props.repository.path),
```

Comment:
Once this method becomes asynchronous, you can move this to `componentDidMount` as you can make that method asynchronous.